students/TacSPx/lesson03/mailroom_modified.py

```diff
@@ -0,0 +1,166 @@
+            ("Jermaine Kearse", [877.33]),
+            ("Marcus Trufant", [1563.23, 1043.87, 1.32]),
+            ("K.J. Wright", [21663.23, 300.87, 100432.0]),
+            ("Curt Warner", [663.23, 300.87, 10432.0]),
+            ]
+
+# Processing  --------------------------------------------------------------- #
+# Task 2:  Menu choices
+# Prompt the user to choose from a menu of 3 actions: “Send a Thank You”, “Create a Report” or “quit”.
+###############################################
+#   "Menu of Choices"
+###############################################
+
+def main():
+    '''
+    Menu of Choices:  Thank You, Report, Create a Report
+    :return:
+    '''
+
+    while (True):
+        print("""
+        \n**** XYZ Nonprofit Charity ****\n
+        Menu of Options
+        1) Send a "Thank You"
+        2) Create a Report
+        3) Exit Program
+        """)
+        str_choice = str(input("Which option would you like to perform? [1 to 3] - "))
+        print()  # added a new line for spacing looks
+        if (str_choice.strip() == '1'):
+            thank_you()
+        elif (str_choice.strip() == '2'):
+            report()
+        elif (str_choice.strip() == '3'):
+            exit_program()
+        else:
+            print("\nPlease choose only 1, 2, or 3!")
+
+###############################################
+#   #1 Send a "Thank You"
+###############################################
+def thank_you():
+    '''
+    Thank You:  Will append donations to existing donors, or adds new donors with donations.
+    User can see list of existing donors by typing 'list'.  Once donation is entered script will create
+    custom thank you letter/ email.
+    :return:
+    '''
+    while (True):
+        donor_name = input("Type 'List' to see all names in Database or\nType First and Last Name: ").title()
+        if (donor_name.lower() == 'list'):
+            # define format row
+            row = "{Name:<18s}".format
+            print("\nDONOR REGISTRY:")
+            donor_sort = (sorted(donor_info))
+            for donor in donor_sort:
+                donor_lst = (row(Name=donor[0]))
+                print(donor_lst)
+            input("Press Enter to Continue")
+            print()
+            continue
+
+        if (donor_name.lower() != 'list'):
+            donor_amount = int(input("Enter Amount of Donation: $"))
+            # if name is in list append dollar amount to donors bucket
+            for i in donor_info:
+                if i[0] == donor_name:
+                    i[1].append(donor_amount)
+                    print(f"'{donor_name}' is in registry added new donation amount!")
+                    break
+                # if name not in list add new name to donor info list
+            else:
+                donor_info.append((donor_name, [donor_amount]))
+                print(f"New donor name '{donor_name}' added to registry!")
+                #print(donor_info) #- for testing
+                break
```

Comment:
you don't need `break` statement here and it actually introduces a bug here


your thank you letter is never shown for new donors